packages/permissionless/errors/account.ts

```diff
@@ -0,0 +1,306 @@
+                "",
+                "Possible solutions:",
+                "• Verify that the sender address was generated deterministically from the initCode. (consider leveraging functions like getSenderAddress)",
+                "• Verify that the factory address in the initCode is correct (the factory address is the first 20 bytes of the initCode)",
+                "• Verify that the initCode is correct.",
+                "",
+                docsPath ? `Docs: ${docsPath}` : ""
+            ].join("\n"),
+            {
+                cause
+            }
+        )
+    }
+}
+
+export class InitCodeDidNotDeploySender extends BaseError {
+    static message = /aa15/
+    override name = "InitCodeDidNotDeploySender"
+    constructor({
+        cause,
+        sender,
+        docsPath
+    }: {
+        cause?: BaseError
+        sender: Address
+        docsPath?: string
+    }) {
+        super(
+            [
+                `Init code did not deploy the sender at the address ${sender}.`,
+                "",
+                "Possible reasons:",
+                "• The initCode factory is not creating an account.",
+                "• The initCode factory is creating an account, but is not implemented correctly as it is not deploying at the sender address",
+                "",
+                "Possible solutions:",
+                "• Verify that the factory address in the initCode is correct (the factory address is the first 20 bytes of the initCode).",
+                "• Verify that the initCode factory is implemented correctly. The factory must deploy the smart account at the sender address.",
+                "",
+                docsPath ? `Docs: ${docsPath}` : ""
+            ].join("\n"),
+            {
+                cause
+            }
+        )
+    }
+}
+
+export class SmartAccountNotDeployed extends BaseError {
+    static message = /aa20/
+    override name = "SmartAccountNotDeployed"
+    constructor({
+        cause,
+        sender,
+        docsPath
+    }: {
+        cause?: BaseError
+        sender: Address
+        docsPath?: string
+    }) {
+        super(
+            [
+                `Smart account ${sender} is not deployed.`,
+                "",
+                "Possible reasons:",
+                "• An initCode was not specified, but the sender address (i.e. the smart account) is not deployed.",
+                "",
+                "Possible solutions:",
+                "• If this is the first transaction by this account, make sure the initCode is included in the user operation.",
+                "• If the smart account is already supposed to be deployed, verify that you have selected the correct sender address for the user operation.",
+                "",
+                docsPath ? `Docs: ${docsPath}` : ""
+            ].join("\n"),
+            {
+                cause
+            }
+        )
+    }
+}
+
+export class SmartAccountDoNotHaveEnoughFunds extends BaseError {
```

Comment:
Do -> Does

Or actually why not call this smth like just:

NotEnoughFunds

Reply:
NotEnoughFunds would not indicate is Smart account has no funds or Paymaster